nautobot/ipam/querysets.py

```diff
@@ -184,6 +190,97 @@ def get_queryset(self):
         """
         return super().order_by("host")
 
+    def string_search(self, search):
+        """
+        Interpret a search string and return useful results.
+        """
+        if not search:
+            return self.none()
+
+        network = self._parse_as_network_string(search)
+        broadcast = self._get_broadcast(network)
+        return self.filter(
+            Q(dns_name__icontains=search)
+            | Q(description__icontains=search)
+            | Q(host__lte=broadcast, host__gte=network.network)  # same as `net_host_contained()`
+        )
+
+    def _parse_as_network_string(self, search):
+        """
+        Attempts to parse a (potentially incomplete) IPAddress and return an IPNetwork.
+        eg: '10.10' should be interpreted as netaddr.IPNetwork('10.10.0.0/16')
+        """
+        version = 4
+
+        try:
+            # Disregard netmask
+            search = search.split("/")[0]
+
+            # Attempt to quickly assess v6
+            if ":" in search:
+                version = 6
+
+            # (IPv6) If the value ends with ":" but it's not "::", make it so.
+            if search.endswith(":") and not self.RE_COLON.match(search):
+                search += ":"
+            # (IPv6) If the value is numeric and > 255, append "::"
+            # (IPv6) If the value is a hextet (e.g. "fe80"), append "::"
+            elif any(
+                [
+                    search.isdigit() and int(search) > 255,
+                    self.RE_HEXTET.match(search),
+                ]
+            ):
+                search += "::"
+                version = 6
+
+            call_map = {
+                4: self._parse_ipv4,
+                6: self._parse_ipv6,
+            }
+            return call_map[version](search)
+
+        except netaddr.core.AddrFormatError:
+            ver_map = {4: "0/32", 6: "::/128"}
+            return netaddr.IPNetwork(ver_map[version])
+
+    def _parse_ipv6(self, value):
+        """IPv6 addresses are 8, 16-bit fields."""
+
+        # Get non-empty octets from search string
+        hextets = value.split(":")
+
+        # Before we normalize, check that final value is a digit.
+        if hextets[-1].alnum():
```

Comment:
Whoops this should be `isalphanum()`
```suggestion
        if hextets[-1].alphanum():
```